clang/TranslationUnit.d

```diff
@@ -74,6 +77,16 @@ struct TranslationUnit
         return DiagnosticVisitor(cx);
     }
 
+    @property DiagnosticSet diagnosticSet ()
+    {
+        return DiagnosticSet(clang_getDiagnosticSetFromTU(cx));
+    }
+
+    @property size_t numDiagnostics ()
```

Comment:
Truth say, I didn't make measurements, but I believe so. It maps to one libclang API call.